drivers/disk/flashdisk.c

```diff
@@ -213,8 +225,10 @@ static int flashdisk_cache_commit(struct flashdisk_data *ctx)
 		return 0;
 	}
 
-	if (flash_erase(ctx->info.dev, ctx->cached_addr, ctx->page_size) < 0) {
-		return -EIO;
+	if (ctx->flash_requires_erase) {
+		if (flash_erase(ctx->info.dev, ctx->cached_addr, ctx->page_size) < 0) {
+			return -EIO;
+		}
 	}
```

Comment:
Show me how code with multi instance Disk Access objects on devices that do not have erase requirement at all, skips this code completely.
Because, yeah emulation may be only that "memset", but that code left in multiple subsystems where there is no need for erase path is extra code, and users pay mostly for Flash, so every byte taken away from user is problematic on many levels.